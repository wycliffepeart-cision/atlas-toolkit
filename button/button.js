import { _ as __decorate } from '../tslib.es6-806de115.js';
import { Template } from '../decorators/template.js';

var ButtonTemplate = "<template>\n  <button part=\"button\" type=\"button\">\n    <slot name=\"prefix\" part=\"prefix\"></slot>\n    <slot part=\"label\"></slot>\n    <slot name=\"suffix\" part=\"suffix\"></slot>\n  </button>\n</template>\n";

const BUTTON_ATTRIBUTES = {
    KIND: 'kind',
    DISABLED: 'disabled',
    BOX_TYPE: 'box-type',
};
let Button = class Button extends HTMLElement {
    static get observedAttributes() {
        return Object.values(BUTTON_ATTRIBUTES);
    }
    attributeChangedCallback(name, _oldValue, newValue) {
        const { DISABLED } = BUTTON_ATTRIBUTES;
        if (name === DISABLED) {
            const isDisabled = newValue !== null && newValue !== 'false';
            this.shadowRoot.querySelector('button').toggleAttribute(DISABLED, isDisabled);
        }
    }
    get disabled() {
        const disabledAttr = this.getAttribute('disabled');
        return disabledAttr !== null && disabledAttr !== 'false';
    }
    set disabled(value) {
        this.toggleAttribute('disabled', value);
    }
};
Button = __decorate([
    Template(ButtonTemplate)
], Button);

export { BUTTON_ATTRIBUTES, Button };
