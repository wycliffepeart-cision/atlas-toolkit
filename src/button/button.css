atlas-button{
    --new: 0;
    --button-reset-border: 0;
    --button-reset-outline: 0;
    --button-reset-decoration: none;
    --button-cursor: pointer;
    --button-position: relative;
    --button-transition: .5s all;
    --button-display: inline-grid;
    --button-box-sixing: border-box;
    --button-line-hight: 20px;
    --button-min-width: 64px;
    --button-border-radius: 6px;
    --button-text-color: #fff;
    --button-outlined: 1px solid #ddd;
    --button-background-transparent: transparent;

    /* Disabled */

    --button-disabled-color: #999999;
    --button-disabled-text-color: #7777777;
    --button-disabled-border-color: #dddddd;
    --button-disabled-background-color: #eeeeee;

    /* Color */

    --button-color-primary: #00607F;
    --button-color-secondary: #fff;
    --button-color-error: #fff;
    --button-color-success: #fff;

    /* Size */

    --button-small-padding-block: 6px;
    --button-small-padding-inline: 10px;
    --button-small-padding: var(--button-small-padding-block) var(--button-small-padding-inline);
    --button-small-font-size: 14px;
    --button-small-line-height: var(--button-line-hight);

    --button-medium-font-size: 16px;
    --button-medium-padding-block: 10px;
    --button-medium-padding-inline: 16px;
    --button-medium-padding: var(--button-medium-padding-block) var(--button-medium-padding-inline);
    --button-medium-line-height: var(--button-line-hight);

    --button-large-padding-block: 14px;
    --button-large-padding-inline: 24px;
    --button-large-padding: var(--button-large-padding-block) var(--button-large-padding-inline);
    --button-large-font-size: 18px;
    --button-large-line-height: var(--button-line-hight);

    /* Variant */

    --button-variant-text-color: #fff;
    --button-variant-text-font-color: #fff;
    --button-variant-text-background-color: transparent;
    --button-variant-text-background-hover-color: #f3f9ff;

    --button-variant-outlined-reset-border: 0;
    --button-variant-outlined-color: #fff;
    --button-variant-outlined-font-color: #fff;
    --button-variant-outlined-background-color: transparent;
    --button-variant-outlined-background-hover-color: #f3f9ff;
    --button-variant-outlined-border-color: var(--button-color-primary);

    --button-variant-contained-text-color: #ffffff;
    --button-variant-contained-background-color: var(--button-color-primary);
    --button-variant-contained-background-hover-color: rgba(0, 96, 127, 0.88);

    /* Adornment */

    --button-adornment-padding-block: 1rem;
    --button-adornment-padding-inline: 1rem
}

atlas-button {
    line-height: 0;
    cursor: pointer;
    display: inline-block;
}

/* Reset */

atlas-button::part(top-adornment),
atlas-button::part(left-adornment),
atlas-button::part(right-adornment),
atlas-button::part(bottom-adornment),
atlas-button::part(main-content) {
    display: block;
}

atlas-button::part(top-adornment),
atlas-button::part(left-adornment),
atlas-button::part(right-adornment),
atlas-button::part(bottom-adornment) {
    display: none;
}

/* Button */

atlas-button::part(button) {
    cursor: var(--button-cursor);
    display: var(--button-display);
    position: var(--button-position);
    color: var(--button-color-primary);
    min-width: var(--button-min-width);
    border: var(--button-reset-border);
    transition: var(--button-transition);
    box-sizing: var(--button-box-sixing);
    outline: var(--button-reset-outline);
    line-height: var(--button-line-hight);
    border-radius: var(--button-border-radius);
    text-decoration: var(--button-reset-decoration);

    align-items: center;
    justify-content: center;
    background-color: transparent;
    grid-template-rows: auto auto auto;
    grid-template-columns: auto auto auto;
}

atlas-button::part(button):hover {
    background-color: #f3f9ff;
}

/* Size */

atlas-button::part(button){
    font-size: var(--button-medium-font-size);
    padding: var(--button-medium-padding);
    border: var(--button-variant-outlined-reset-border);
    color: var(--button-variant-contained-text-color);
    background-color: var(--button-variant-contained-background-color);
}

atlas-button::part(button):hover {
    background-color: var(--button-variant-contained-background-hover-color);
}

atlas-button[size="small"]::part(button) {
    font-size: var(--button-small-font-size);
    padding: var(--button-small-padding);
}

atlas-button[variant="outlined"]::part(button),
atlas-button[size="small"][variant="outlined"]::part(button){
    padding-left: calc(var(--button-small-padding-inline) - 1px);
    padding-right: calc(var(--button-small-padding-inline) - 1px);
}

atlas-button[size="medium"]::part(button) {
    font-size: var(--button-medium-font-size);
    padding: var(--button-medium-padding);
}

atlas-button[size="medium"][variant="outlined"]::part(button){
    padding-left: calc(var(--button-medium-padding-inline) - 1px);
    padding-right: calc(var(--button-medium-padding-inline) - 1px);
}

atlas-button[size="large"]::part(button) {
    font-size: var(--button-large-font-size);
    padding: var(--button-large-padding);

}

atlas-button[size="large"][variant="outlined"]::part(button){
    padding-left: calc(var(--button-large-padding-inline) - 1px);
    padding-right: calc(var(--button-large-padding-inline) - 1px);
}

/* Variant */

atlas-button[variant="text"]::part(button) {
    color: var(--button-color-primary);
    background-color: var(--button-variant-outlined-background-color);
}

atlas-button[variant="text"]::part(button):hover {
    background-color: var(--button-variant-text-background-hover-color);
}

atlas-button[variant="text"][disabled]::part(button){
    color: var(--button-disabled-color);
    background-color: var(--button-background-transparent);
}

atlas-button[variant="outlined"]::part(button) {
    color: var(--button-color-primary);
    border: 1px solid var(--button-variant-outlined-border-color);
    background-color: var(--button-variant-outlined-background-color);
}

atlas-button[variant="outlined"]::part(button):hover {
    background-color: var(--button-variant-outlined-background-hover-color);
}

atlas-button[variant="outlined"][disabled]::part(button){
    color: var(--button-disabled-color);
    border: 1px solid var(--button-disabled-border-color);
    background-color: var(--button-background-transparent);
}

/*TODO: review*/
atlas-button[variant="contained"] atlas-typography::part(span) {
    color: inherit;
    font-weight: 500;
    line-height: inherit;
    font-size: 1rem;
}

atlas-button[variant="contained"]::part(button) {
    border: var(--button-variant-outlined-reset-border);
    color: var(--button-variant-contained-text-color);
    background-color: var(--button-variant-contained-background-color);
}

atlas-button[variant="contained"][disabled]::part(button){
    color: var(--button-disabled-text-color);
    border: 1px solid var(--button-disabled-border-color);
    background-color: var(--button-disabled-background-color);
}

atlas-button[variant="contained"]::part(button):hover{
    background-color: var(--button-variant-contained-background-hover-color);
}

/* Top Adornment */

atlas-button::part(top-adornment) {
    display: flex;
    grid-column: 1/-1;
}

atlas-button:has([slot="top-adornment"])::part(top-adornment){
    margin-bottom: .5rem;
}

atlas-button[bottom-adornment="center"]::part(bottom-adornment),
atlas-button[top-adornment="center"]::part(top-adornment) {
    align-items: center;
    justify-content: center;
}

atlas-button[top-adornment="end"]::part(top-adornment) {
    align-items: end;
    justify-content: end;
}

/* Left Adornment */

atlas-button::part(left-adornment) {
    grid-row: 2/3;
    grid-column: 1/2;
    display: flex;
    align-items: center;
    justify-content: center;
}

atlas-button:has([slot="left-adornment"])::part(left-adornment) {
    margin-right: 8px;
}

/* Main Content */

atlas-button::part(main-content) {
    display: flex;
    grid-column: 2/3;
    align-items: center;
    vertical-align: center;
    justify-content: center;
    letter-spacing: 0.02857em;
}

/* Right Adornment */

atlas-button::part(right-adornment) {
    display: flex;
    grid-column: 3/-1;
    align-items: center;
    justify-content: center;
}

atlas-button:has([slot="right-adornment"])::part(right-adornment) {
    margin-left: 8px;
}

/* Bottom Adornment*/

atlas-button::part(bottom-adornment) {
    grid-column: 1/-1;
    grid-row: 3/-1;
    display: flex;
}

atlas-button:has([slot="bottom-adornment"])::part(bottom-adornment){
    margin-top: .5rem;
}

/* Common */

atlas-button[disabled]{
    opacity: 0.5;
    color: #999999;
    pointer-events: none !important;
}