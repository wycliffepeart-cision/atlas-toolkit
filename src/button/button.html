<template>
  <button part="button" type="button">
    <slot name="prefix" part="prefix"></slot>
    <slot part="label"></slot>
    <slot name="suffix" part="suffix"></slot>
  </button>
</template>
