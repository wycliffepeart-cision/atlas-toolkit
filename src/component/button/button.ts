import ButtonTemplate from './button.html';
import { Template } from '../../decorators/template';

export const BUTTON_ATTRIBUTES = {
  KIND: 'kind',
  DISABLED: 'disabled',
  BOX_TYPE: 'box-type',
};

@Template(ButtonTemplate)
export class Button extends HTMLElement {
  static get observedAttributes() {
    return Object.values(BUTTON_ATTRIBUTES);
  }

  attributeChangedCallback(name, _oldValue, newValue) {
    const { DISABLED } = BUTTON_ATTRIBUTES;

    if (name === DISABLED) {
      const isDisabled = newValue !== null && newValue !== 'false';
      this.shadowRoot.querySelector('button').toggleAttribute(DISABLED, isDisabled);
    }
  }

  get disabled() {
    const disabledAttr = this.getAttribute('disabled');
    return disabledAttr !== null && disabledAttr !== 'false';
  }

  set disabled(value) {
    this.toggleAttribute('disabled', value);
  }
}

if (!customElements.get('atlas-button')) customElements.define('atlas-button', Button);

