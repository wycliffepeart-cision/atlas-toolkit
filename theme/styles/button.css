@layer atlas {
	/* Variables */
	atlas-button {
		--default-font-size: 1em;
		--default-font-weight: 400;
		--default-line-heigh: 2.5ch;
		--default-background-color: unset;
		--default-disabled-text-color: var(--atlas-color-primary-n300);

		--large-padding-top-bottom: 0.7em;
		--large-padding-left-right: 1.25em;

		--medium-line-heigh: 1.25rem;
		--medium-font-size: 0.875em;
		--medium-padding-top-bottom: 0.375em;

		--primary-active-background: var(--atlas-color-primary-b300);

		--secondary-border-color: var(--atlas-color-primary-b500);
		--secondary-color: var(--atlas-color-primary-b500);
		--secondary-hover-background-color: var(--atlas-color-primary-b25);

		--tertiary-text-color: var(--atlas-color-primary-n400);
		--tertiary-focus-shadow: var(--atlas-color-primary-n200);
		--tertiary-focus-outline: var(--atlas-color-primary-n200);
		--tertiary-border-color: var(--atlas-color-primary-n200);
		--tertiary-active: var(--atlas-color-primary-n100);
		--tertiary-background-hover: var(--atlas-color-primary-n25);

		--icon-default-color: var(--atlas-color-primary-n300);
		--icon-active-color: var(--atlas-color-primary-b500);
		--icon-hover-color: var(--atlas-color-primary-o500);

		--text-color: var(--atlas-color-primary-b400);
		--text-color-active: var(--atlas-color-primary-b400);
		--text-color-hover: var(--atlas-color-primary-b300);
		--text-disabled-color: var(--atlas-color-primary-n100);
		--text-space-between: 0.65em;
		--default-disabled-color: var(--atlas-color-primary-n100);

		--box-secondary-active: var(--atlas-color-primary-b50);
		--box-disabled-background: var(--atlas-color-primary-n50);
		--box-focus-shadow-blur-radius: 0.5em;
		--box-focus-shadow-spread-radius: 0.06em;
		--box-focus-shadow-color: var(--atlas-button-primary-background-color);
		--box-focus-shadow: 0 0 var(--box-focus-shadow-blur-radius)
			var(--box-focus-shadow-spread-radius) var(--box-focus-shadow-color);
		--box-focus-shadow-tertiary: 0 0 var(--box-focus-shadow-blur-radius)
			var(--box-focus-shadow-spread-radius) var(--tertiary-focus-shadow);

		--default-border-radius: 0.25em;
		--default-border-size: 0.04em;
		--default-focus-outline: 0.1em;
	}

	/* Shared styles */
	atlas-button {
		background-color: var(--default-background-color);
		display: flex;
		font-family: inherit;
		font-weight: var(--default-font-weight);
		font-size: var(--default-font-size);
		height: 100%;
		line-height: var(--default-line-heigh);
		max-width: max-content;
		min-width: max-content;
	}

	atlas-button[minor-action]:not([minor-action="false"]) {
		line-height: var(--medium-line-heigh);
		font-size: var(--medium-font-size);
		padding: var(--medium-padding-top-bottom) 0;
		max-width: max-content;
		min-width: max-content;
	}

	atlas-button::part(button) {
		align-items: center;
		border: none;
		display: inline-flex;
		gap: var(--text-space-between);
		padding: var(--large-padding-top-bottom) var(--large-padding-left-right);

		background-color: inherit;
		border-radius: inherit;
		color: inherit;
		cursor: inherit;
		font-size: inherit;
		font-family: inherit;
	}

	atlas-button:hover {
		cursor: pointer;
	}

	atlas-button[disabled]:not([disabled="false"]) {
		cursor: unset;
		pointer-events: none;
		border-color: var(--box-disabled-background);
	}

	/* Box */
	atlas-button[box-type][disabled]:not([disabled="false"]) {
		background-color: var(--box-disabled-background);
		color: var(--default-disabled-text-color);
		border-color: var(--box-disabled-background);
	}

	atlas-button,
	atlas-button[kind="box"],
	atlas-button[box-type] {
		border-radius: var(--default-border-radius);
	}

	/* Icon button */
	atlas-button[kind="icon"] atlas-icon {
		color: inherit;
	}

	atlas-button[kind="icon"] {
		background-color: unset;
		outline: 0;
		color: var(--icon-default-color);
		border: none;
	}

	atlas-button[kind="icon"]:hover {
		cursor: pointer;
		color: var(--icon-hover-color);
		background-color: unset;
	}

	atlas-button[kind="icon"]::part(button) {
		padding: 0;
	}

	atlas-button[kind="icon"]::part(prefix),
	atlas-button[kind="icon"]::part(suffix) {
		display: none;
	}

	atlas-button[kind="icon"]:active {
		color: var(--icon-active-color);
	}

	atlas-button[kind="icon"][disabled]:not([disabled="false"]) {
		background: transparent;
		color: var(--default-disabled-color);
		border: none;
	}

	/* Text */
	atlas-button[kind="text"] {
		flex-direction: column;
		background-color: unset;
		padding: 0;
		color: var(--text-color);
		outline: 0;
		border: none;
	}

	atlas-button:where([kind="text"]) {
		margin-block: auto;
	}

	atlas-button[kind="text"]::part(button) {
		padding: 0;
	}

	atlas-button[kind="text"]:hover {
		color: var(--text-color-hover);
		background-color: unset;
	}

	atlas-button[kind="text"]:active {
		color: var(--text-color-active);
	}

	atlas-button[kind="text"][disabled]:not([disabled="false"]) {
		background-color: unset;
		color: var(--text-disabled-color);
		border: none;
	}

	/* Primary */
	atlas-button,
	atlas-button[box-type="primary"] {
		background-color: var(--atlas-button-primary-background-color);
		color: var(--atlas-button-primary-color);
		border: var(--default-border-size) solid var(--atlas-button-primary-background-color);
	}

	atlas-button[disabled]:not([disabled="false"]),
	atlas-button[box-type="primary"][disabled]:not([disabled="false"]) {
		background-color: var(--atlas-button-primary-disabled-background-color);
		border-color: var(--atlas-button-primary-disabled-background-color);
	}
	atlas-button:hover,
	atlas-button[box-type="primary"]:hover {
		background-color: var(--atlas-button-hover-background-color);
		border-color: var(--atlas-button-hover-background-color);
	}

	atlas-button:active,
	atlas-button[box-type="primary"]:active {
		background-color: var(--primary-active-background);
		border-color: var(--primary-active-background);
	}

	/* Secondary */
	atlas-button[box-type="secondary"] {
		background-color: transparent;
		border: var(--default-border-size) solid var(--secondary-border-color);
		color: var(--secondary-color);
	}

	atlas-button[box-type="secondary"]:hover {
		background-color: var(--secondary-hover-background-color);
	}

	atlas-button[box-type="secondary"]:active {
		background-color: var(--box-secondary-active);
	}

	/* Tertiary */
	atlas-button[box-type="tertiary"] {
		background-color: transparent;
		color: var(--tertiary-text-color);
		border: var(--default-border-size) solid var(--tertiary-border-color);
	}

	atlas-button[box-type="tertiary"]:hover {
		background-color: var(--tertiary-background-hover);
	}
	atlas-button[box-type="tertiary"]:active {
		background-color: var(--tertiary-active);
	}

	/* Focus */
	atlas-button::part(button):focus-visible,
	atlas-button[box-type="primary"]::part(button):focus-visible,
	atlas-button[box-type="secondary"]::part(button):focus-visible {
		box-shadow: var(--box-focus-shadow);
		border-radius: var(--default-border-radius);
		outline: var(--default-focus-outline) solid var(--atlas-button-primary-background-color);
	}

	atlas-button[box-type="tertiary"]::part(button):focus-visible {
		box-shadow: var(--box-focus-shadow-tertiary);
		border-radius: var(--default-border-radius);
		outline: var(--default-focus-outline) solid var(--tertiary-focus-outline);
	}

	atlas-button[kind="icon"]::part(button):focus-visible,
	atlas-button[kind="text"]::part(button):focus-visible {
		box-shadow: none;
		border-radius: 0;
		outline: 0;
	}

	/* Generic Button */
	atlas-button[kind="generic"] {
		border: none;
		display: inline-block;
		outline: none;
		color: inherit;
		max-width: inherit;
		min-width: inherit;
		line-height: normal;
		background-color: inherit;
	}

	atlas-button[kind="generic"]::part(button) {
		margin: 0;
		padding: 0;
	}

	atlas-button::part(prefix),
	atlas-button::part(suffix) {
		line-height: 100%;
	}
}
