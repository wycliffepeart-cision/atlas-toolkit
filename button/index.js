import { Button } from './button.js';
import '../tslib.es6-806de115.js';
import '../decorators/template.js';

if (!customElements.get('atlas-button'))
    customElements.define('atlas-button', Button);
