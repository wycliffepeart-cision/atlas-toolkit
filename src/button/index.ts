import { Button } from './button';
import '../../theme/styles/button.css';

if (!customElements.get('atlas-button')) customElements.define('atlas-button', Button);
