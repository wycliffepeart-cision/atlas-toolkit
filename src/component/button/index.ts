import { Button } from './button';

if (!customElements.get('atlas-button')) customElements.define('atlas-button', Button);
