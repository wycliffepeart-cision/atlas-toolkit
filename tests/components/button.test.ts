import { fireEvent } from '@testing-library/dom';
import '../../src/component/button/button';

const ATLAS_BUTTON_TAG = 'atlas-button';
const getShadowDomContent = (element) => element.shadowRoot.innerHTML;

const setupElement = async () => {
  await customElements.whenDefined(ATLAS_BUTTON_TAG);
  const buttonElement = document.createElement(ATLAS_BUTTON_TAG);
  document.body.appendChild(buttonElement);

  return buttonElement;
};

afterEach(() => {
  const element = document.querySelector(ATLAS_BUTTON_TAG);
  if (element) {
    element.remove();
  }
});

describe('Atlas Button', () => {
  it('is correctly appended in the DOM', async () => {
    const buttonElement = await setupElement();

    (expect(buttonElement) as any).toBeInTheDocument();
  });

  it('is capable of being selected', async () => {
    await setupElement();

    const selectedElement = document.querySelector(ATLAS_BUTTON_TAG);
    (expect(selectedElement) as any).toBeInTheDocument();
  });

  it('has the correct content when created', async () => {
    const buttonElement = await setupElement();

    (expect(buttonElement) as any).toContainHTML('<atlas-button></atlas-button>');
    (expect(buttonElement) as any).toBeEmptyDOMElement();
    (expect(getShadowDomContent(buttonElement)) as any).toMatchSnapshot();
  });

  it('Should be interactable', async () => {
    const buttonElement = await setupElement();
    const mockClickEvent = jest.fn();
    const mockFocusEvent = jest.fn();

    buttonElement.addEventListener('click', mockClickEvent);
    buttonElement.addEventListener('focus', mockFocusEvent);

    fireEvent.click(buttonElement);
    fireEvent.focus(buttonElement);

    (expect(buttonElement) as any).toBeEnabled();
    (expect(mockClickEvent) as any).toHaveBeenCalledTimes(1);
    (expect(mockFocusEvent) as any).toHaveBeenCalledTimes(1);
  });

  it('Should be disabled when disabled attribute is set', async () => {
    const buttonElement = await setupElement();

    buttonElement.setAttribute('disabled', '');
    (expect(buttonElement) as any).toBeDisabled();

    buttonElement.removeAttribute('disabled');
    (expect(buttonElement) as any).toBeEnabled();
  });
});
