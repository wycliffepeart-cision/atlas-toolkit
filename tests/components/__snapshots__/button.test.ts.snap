// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Atlas Button has the correct content when created 1`] = `
<button part="button"
        type="button"
>
  <slot name="prefix"
        part="prefix"
  >
  </slot>
  <slot part="label">
  </slot>
  <slot name="suffix"
        part="suffix"
  >
  </slot>
</button>
`;
